nextcord/gateway.py

```diff
@@ -395,7 +396,9 @@ async def from_client(
         await ws.resume()
         return ws
 
-    def wait_for(self, event, predicate, result=None):
+    def wait_for(
+        self, event: str, predicate: Callable, result: Optional[Callable] = None
```

Comment:
Be more specific with the type hints for the parameters `predicate` and `result`